components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -314,7 +314,7 @@ public class OAuth2Util {
     private static final Log diagnosticLog = LogFactory.getLog("diagnostics");
     private static final String INTERNAL_LOGIN_SCOPE = "internal_login";
     public static final String JWT = "JWT";
-    private static long timestampSkew = OAuthServerConfiguration.getInstance().getTimeStampSkewInSeconds() * 1000;
+    public static long timestampSkew = OAuthServerConfiguration.getInstance().getTimeStampSkewInSeconds() * 1000;
```

Comment:
Is this required when there is a public method to get skew time? An additional option would be adding another method into OAuthServerConfiguration to return skew time in milliseconds.

Reply:
1. Nonstatic + calculated/derived `OAuthServerConfiguration.getInstance().getTimeStampSkewInSeconds() * 1000` is costly when compared to already available static` OAuth2Util.timestampSkew`. Therefore, there was no other alternative.
2. I can use a public getter method to enforce encapsulation so no-other class can change the value (since there would be no setter method) or better alternative would be to make it a `constant` by making it unassignable after first time initialization besides, there would not be any impact as already it was already private (used within the class). `final` keyword would do than a getter. WDYT?

Reply:
1. Nonstatic + calculated/derived `OAuthServerConfiguration.getInstance().getTimeStampSkewInSeconds() * 1000` is costly when compared to already available static` OAuth2Util.timestampSkew`. Therefore, there was no other alternative but to use the stored `static` value rather than using the `public` getter and deriving (*1000) to the value each time query runs.
2. Better alternative would be to make it a `constant` by making it unassignable after one time initialization, besides there would not be any impact as it was already `private` (used within the class). Adding `final` keyword would do than a `public` getter method (i.e `public static final long timestampSkew`). WDYT?